bazel/cc_wrapper.py

```diff
@@ -94,5 +96,72 @@ def main():
   os.execv(compiler, [compiler] + argv)
 
 
+def new_modify_driver_command_line_and_execute(flagfile_path, new_flagfile_path, new_flagfile_fd):
+  flagfile_contents = open(flagfile_path, "r").readlines()
+
+  # Detect if we're building for C++ or vanilla C.
+  if any(map(new_is_cpp_flag, flagfile_contents)):
+    compiler = envoy_real_cxx
+    # Append CXXFLAGS to all C++ targets (this is mostly for dependencies).
+    argv = shlex.split(envoy_cxxflags)
+  else:
+    compiler = envoy_real_cc
+    # Append CFLAGS to all C targets (this is mostly for dependencies).
+    argv = shlex.split(envoy_cflags)
+
+  # Either:
+  # a) remove all occurrences of -lstdc++ (when statically linking against libstdc++),
+  # b) replace all occurrences of -lstdc++ with -lc++ (when linking against libc++).
+  if "-static-libstdc++\n" in flagfile_contents or "-stdlib=libc++" in envoy_cxxflags:
```

Comment:
As mentioned above, much better to strip the whitespace so we don't have to reason about this.

Reply:
Done.